### PR TITLE
Add support to pass in * to select all namespaces to backup and update rule to take container name

### PR DESCRIPTION
**Pull request by disrani-px**
**What this PR does / why we need it**:
Add support to pass in * to select all namespaces to backup

Update Rule spec to take in container name
This is required since a pod could have multiple containers and we
need to know which one we should run the commands in

**Does this PR change a user-facing CRD or CLI?**:

Adds container field to Rule spec

**Is a release note needed?**:

```release-note
* Added support to specify * as a wildcard to select all namespaces in ApplicationBackup spec
* Added container name in Rule spec. This is required if you have multiple containers in a pod and want to select a container to run the command in.

```

**Does this change need to be cherry-picked to a release branch?**:

No

### CODE REVIEW

File: pkg/applicationmanager/controllers/applicationbackup.go [hunk 1]
```diff
@@ -156,13 +158,34 @@ func (a *ApplicationBackupController) performRuleRecovery() error {
...
 		backup.Status.TriggerTimestamp = backup.CreationTimestamp
+		updated = true
+	}
+	return updated
+}
+
+func (a *ApplicationBackupController) updateWithAllNamespaces(backup *stork_api.ApplicationBackup) error {
+	namespaces, err := core.Instance().ListNamespaces(nil)
+	if err != nil {
+		return fmt.Errorf("error updating with all namespaces for wildcard: %v", err)
```
Comment (prashanthpx) [inline pkg/applicationmanager/controllers/applicationbackup.go:177]: nit:  as this is listing, should `updating` be changed to `listing`

File: pkg/rule/rule.go [hunk 2]
```diff
@@ -248,11 +249,13 @@ func ExecuteRule(
...
 		backgroundCommandTermChan := make(chan bool, 1)
 		backgroundPodListChan := make(chan v1.Pod)
-		go cmdTerminationWatcher(backgroundPodListChan, backgroundCommandTermChan, owner, taskID.String())
+		var container string
+		go cmdTerminationWatcher(backgroundPodListChan, &container, backgroundCommandTermChan, owner, taskID.String())
 
 		// backgroundActionPresent is used to track if there is atleast one background action
 		backgroundActionPresent := false
 		for _, item := range rule.Rules {
+			container = item.Container
```
Comment (prashanthpx) [inline pkg/rule/rule.go:258]: To understand, if the user doesn't provide container input in the Rule spec, would it fall back to running on all containers?

File: pkg/applicationmanager/controllers/applicationbackup.go (hunk 1 above)
Comment (disrani-px) [reply pkg/applicationmanager/controllers/applicationbackup.go:177]: The function is trying to update the backup spec with all the namespaces

File: pkg/rule/rule.go (hunk 2 above)
Comment (disrani-px) [reply pkg/rule/rule.go:258]: No, it'll throw an error, which is the current behavior. We can't run the command in all the containers

**Pull request merged**